algorithms/DepthFirstSearch.java

```diff
@@ -0,0 +1,46 @@
+package algorithms;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Stack;
+
+public class DepthFirstSearch extends Algorithm {
+    List<Node> visitedNodes = new ArrayList<Node>();
+
+    public DepthFirstSearch(Node startNode, Node goalNode, Collection<Node> borders) {
+        super(startNode, goalNode, borders);
+        openList = new Stack<Node>();        
+    }
+
+    @Override
+    public void findPath() {
+		if (!this.openList.isEmpty()) {
+            Node currentNode = ((Stack<Node>) openList).pop();
```

Comment:
I don't think it is good to abstract the openList in the `Algorithm` class. You need a different data structure for the different algorithms.